### PR TITLE
Use latest istio/api

### PR DESCRIPTION
**Pull request by guptasu**
* Use latest istio/api repo SHA. Latest version has correct .pb.go files (originally missing that was causing `go test` to not work and also circle-ci)

* Latest version on istio/api also enforced istio/istio to use the BUILD rules from Istio/api/*BUILD files instead of defining a new one in mixer/istio_api.bzl
(Deleted the code from istio_api.bzl)

* Make fmt

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: bin/gazelle [hunk 1]
```diff
@@ -17,11 +17,11 @@ fi
...
-          -e 's|@io_istio_api//mixer/v1/config/descriptor:go_default_library|@io_istio_api//:mixer/v1/config/descriptor|g' \
-          -e 's|@io_istio_api//mixer/v1:go_default_library|@io_istio_api//:mixer/v1|g' \
+          -e 's|@io_istio_api//mixer/v1/config/descriptor:descriptor|@io_istio_api//:mixer/v1/config/descriptor|g' \
+          -e 's|@io_istio_api//:mixer/v1|@io_istio_api//:mixer/v1|g' \
           -e 's|@io_istio_api//mixer/v1/config/client:go_default_library|@io_istio_api//:mixer/v1/config/client|g' \
           -e 's|@io_istio_api//mixer/v1/template:go_default_library|@io_istio_api//:mixer/v1/template|g' \
-          -e 's|@io_istio_api//proxy/v1/config:go_default_library|@io_istio_api//:proxy/v1/config|g'\
-          -e 's|@io_istio_api//broker/v1/config:go_default_library|@io_istio_api//:broker/v1/config|g'\
+          -e 's|@io_istio_api//proxy/v1/config:go_default_library|@io_istio_api//proxy/v1/config:go_default_library|g'\
+          -e 's|@io_istio_api//broker/v1/config:go_default_library|@io_istio_api//broker/v1/config|g'\
```
Comment (ayj) [inline bin/gazelle]: Looks like we should be able to delete most (all?) of these substitutions now that we're using the default generated targets?

File: mixer/pkg/config/proto/BUILD [hunk 2]
```diff
@@ -7,12 +7,12 @@ go_library(
...
         "combined.go",
-        "@io_istio_api//:mixer/v1/config_fixed",  # keep
+        "@io_istio_api//mixer/v1/config:config_fixed",  # keep
     ],
     deps = [
         "@com_github_gogo_protobuf//proto:go_default_library",
         "@com_github_gogo_protobuf//sortkeys:go_default_library",
         "@com_github_gogo_protobuf//types:go_default_library",
-        "@io_istio_api//:mixer/v1/config/descriptor",  # keep
+        "@io_istio_api//mixer/v1/config/descriptor:descriptor",  # keep
```
Comment (ayj) [inline mixer/pkg/config/proto/BUILD]: I think `keep` here and elsewhere is no longer needed?

File: bin/gazelle (hunk 1 above)
Comment (guptasu) [reply bin/gazelle]: why did we added this subs ?

File: mixer/pkg/config/proto/BUILD (hunk 2 above)
Comment (guptasu) [reply mixer/pkg/config/proto/BUILD]: What does the `#keep` do ?

Comment (kyessenov) [reply mixer/pkg/config/proto/BUILD]: It prevents gazelle from overwriting. Try to remove those and run gazelle, and see what happens.

Comment (guptasu) [reply mixer/pkg/config/proto/BUILD]: D

**Pull request merged**